build/build.go

```diff
@@ -55,6 +56,7 @@ func Build(buildFilePath, gitRootDir string) (*slsa.Provenance, error) {
 			return nil, fmt.Errorf("couldn't fetch sources from %s: %v", buildConfig.Repo, err)
 		}
 		log.Printf("Fetched the repo into %q. See %q for any error logs.", repoInfo.RepoRoot, repoInfo.Logs)
+		tempFolder = repoInfo.RepoRoot
```

Comment:
Alternatively, you could add a method `repoInfo.Cleanup()` on this object, and call `defer` on it here. Especially since we still want the cleanup to happen even if the function does not run until the end, right? WDYT?